### PR TITLE
Vectorized block diagonal mat inversion

### PR DESCRIPTION
**Pull request by OmarDuran**
## Proposed changes

This PR refactor and introduce a vectorized solution for the function `invert_diagonal_blocks_python` in file `matrix_operations.py`. The function is 2x faster than numba function for a large block diagonal matrix with irregular block sizes.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [x] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [ ] The documentation is up-to-date.
- [ ] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: src/porepy/numerics/linalg/matrix_operations.py [hunk 1]
```diff
@@ -567,19 +567,24 @@ def invert_diagonal_blocks_python(a: sps.spmatrix, sz: np.ndarray) -> np.ndarray
...
+        row, col = a.nonzero()
+        idx_blocks = np.cumsum([0] + list(sz))
+        idx_nnz = np.searchsorted(row, idx_blocks)
+
+        def sub_block(ib):
+            lr = row[idx_nnz[ib] : idx_nnz[ib + 1]] - idx_blocks[ib]
+            lc = col[idx_nnz[ib] : idx_nnz[ib + 1]] - idx_blocks[ib]
+            dense_block = np.zeros((sz[ib], sz[ib]))
+            dense_block[lr, lc] = a.data[idx_nnz[ib] : idx_nnz[ib + 1]]
+            return dense_block
```
Comment (keileg) [inline src/porepy/numerics/linalg/matrix_operations.py:583]: Please document this part.

File: src/porepy/numerics/linalg/matrix_operations.py [hunk 2]
```diff
@@ -694,15 +699,15 @@ def inv_python(indptr, ind, data, sz):
 
     # Remove blocks of size 0
     s = s[s > 0]
-    # Variable to check if we have tried and failed with numba
-    if method == "numba" or method is None:
+    # Select python vectorized function
+    if method == "python" or method is None:
```
Comment (keileg) [inline src/porepy/numerics/linalg/matrix_operations.py:703]: I agree with making this the default, but to do so properly, we should also change the default value of the inverter in mpfa, mpsa (and possibly Biot).

File: src/porepy/numerics/linalg/matrix_operations.py [hunk 3]
```diff
@@ -694,15 +699,15 @@ def inv_python(indptr, ind, data, sz):
...
+        inv_vals = invert_diagonal_blocks_python(mat, s)
```
Comment (keileg) [inline src/porepy/numerics/linalg/matrix_operations.py:704]: (emtpy comment, just to start a formal review). Disregard.

**Review by keileg — COMMENTED** [review]
Control question: You have verified that the inverted matrices are correct?

Given the performance results, I think we should make this into the default inverter. This will require changing parameter values in mpfa, mpsa and biot. A repository-wide search for 'inverter' may be needed.

File: src/porepy/numerics/linalg/matrix_operations.py (hunk 1 above)
Comment (OmarDuran) [reply src/porepy/numerics/linalg/matrix_operations.py]: I have documented the function.

File: src/porepy/numerics/linalg/matrix_operations.py (hunk 2 above)
Comment (OmarDuran) [reply src/porepy/numerics/linalg/matrix_operations.py:724]: I have updated the default on those files.

**Pull request merged**